File: app/src/main/java/com/arthur/gloria/logired/features/trip/available/data/repository/AvailableTripsRepositoryImpl.kt
package com.arthur.gloria.logired.features.trip.available.data.repository

import com.arthur.gloria.logired.core.network.ApiService
import com.arthur.gloria.logired.core.network.model.Trip
import com.arthur.gloria.logired.features.trip.available.domain.repository.AvailableTripsRepository
import javax.inject.Inject

class AvailableTripsRepositoryImpl @Inject constructor(
    private val api: ApiService
) : AvailableTripsRepository {

    override suspend fun getAvailableTrips(city: String): Result<List<Trip>> {
        return try {
            val response = api.getTripsByCity(city)
            if (response.isSuccessful && response.body() != null) {
                Result.success(response.body()!!.rides ?: emptyList())
            } else {
                Result.failure(Exception("Error al obtener viajes disponibles: ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(Exception("Error de conexión: ${e.message}"))
        }
    }

    override suspend fun acceptTrip(tripId: Int): Result<Unit> {
        return try {
            val response = api.acceptTrip(tripId)
            if (response.isSuccessful) {
                Result.success(Unit)
            } else {
                Result.failure(Exception("Error al aceptar viaje: ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(Exception("Error de conexión: ${e.message}"))
        }
    }
}

File: lib/features/trip/available/data/repository/available_trips_repository_impl.dart
import '../../../../../core/network/api_service.dart';
import '../../../../../core/network/model/models.dart';
import '../../domain/available_trips_repository.dart';

class AvailableTripsRepositoryImpl implements AvailableTripsRepository {
  final ApiService _api;

  AvailableTripsRepositoryImpl(this._api);

  @override
  Future<List<Trip>> getAvailableTrips(String city) async {
    final response = await _api.getAvailableTrips();
    final data = response.data;
    List<dynamic> list;
    if (data is List) {
      list = data;
    } else if (data is Map) {
      list =
          (data['rides'] ?? data['data'] ?? data['trips'] ?? []) as List? ?? [];
    } else {
      list = [];
    }
    return list.map((e) => Trip.fromJson(e as Map<String, dynamic>)).toList();
  }

  @override
  Future<void> acceptTrip(int tripId) async {
    await _api.acceptTrip(tripId);
  }
}
